### DISCUSSION TITLE
[Feature Request] Adding an option for Switching Rectangle Pro to control the active application instead of the window under the curso

### DISCUSSION THREAD

**Discussion by bigplayer-ai** [rxhanson/RectanglePro-Community#593] [Ideas]
Adding an option to switch Rectangle Pro to control the **active application** instead of the **window under the cursor** would significantly enhance productivity and accessibility, especially for users with multiple screens. Currently, fullscreen applications obscure other apps, even though background apps remain active. Allowing Rectangle Pro to control the active rather than the window under the mouse pointer would resolve this limitation.

It would be particularly beneficial if this functionality were implemented for keyboard shortcuts only. This feature would provide a substantial productivity boost by making it easier to manage multiple applications efficiently.

As you can see ChatGPT is active but I can't control it via Rectangle Pro keyboard shortcuts because I am on fullscreen on the external monitor, and ChatGPT is on different space on the external monitor space. I would like to control ChatGPT to move to the different monitor using my rectangle pro keyboard shortcuts.

![ChatGPT is active but not on the foreground (Different monitor)](https://github.com/user-attachments/assets/f4518055-c169-4a49-b465-bfe4b684dc1e)

![Rectangle Pro Relevant Settings](https://github.com/user-attachments/assets/3a663e7a-96a6-457b-b775-765ca1829b64)

**Comment by rxhanson** [comment]
Thanks for the feedback. Currently, keyboard shortcuts only work on the frontmost/active window, and not on the window under the cursor. If I understand correctly, what you're experiencing is that Rectangle Pro is unable to manipulate a full screened windows, and this is a limitation of macOS. Building in something that would un-full-screen, move a window, then full-screen again would depend on timing and would be pretty fragile, which makes it undesirable to add in. If I'm misunderstanding, then I'll need a little more information to know exactly what's going wrong.
